frontend/src/pages/patients/PatientsPage.tsx

```diff
@@ -100,9 +246,146 @@ export const PatientsPage: React.FunctionComponent = (props) => {
             ],
           } as SampleWhere
         }
+        customFilterUI={
+          <>
+            <Col md="auto" className="mt-1">
+              <div className="vr"></div>
+            </Col>
+
+            <Col md="auto" className="mt-1">
+              <Form>
+                <Form.Check
+                  type="switch"
+                  id="custom-switch"
+                  label="PHI-enabled"
+                  checked={phiEnabled}
+                  onChange={(e) => {
+                    setPhiEnabled(e.target.checked);
+                  }}
+                />
+              </Form>
+            </Col>
+
+            <Col md="auto" style={{ marginLeft: -15 }}>
+              <Tooltip
+                title={
+                  <span style={{ fontSize: 12 }}>
+                    When enabled, you can search for patients by either their
+                    MRN, CMO Patient ID, or DMP Patient ID. The results will
+                    include an additional column with the patient's MRN.
+                  </span>
+                }
+              >
+                <InfoIcon style={{ fontSize: 18, color: "grey" }} />
+              </Tooltip>
+            </Col>
+          </>
+        }
+        setCustomFilterVals={setPatientIdsTriplets}
+        handleSearch={handleSearch}
+        searchVal={searchVal}
+        setSearchVal={setSearchVal}
+        inputVal={inputVal}
+        setInputVal={setInputVal}
+        showDownloadModal={showDownloadModal}
+        setShowDownloadModal={setShowDownloadModal}
+        handleDownload={() => {
+          setAlertModal({
+            show: true,
+            ...phiWarning,
+          });
+          setShowDownloadModal(true);
+        }}
+      />
+
+      <AlertModal
+        show={alertModal.show}
+        onHide={() => {
+          setAlertModal({ ...alertModal, show: false });
+        }}
+        title={alertModal.title}
+        content={alertModal.content}
       />
     </>
   );
-};
+}
 
-export default PatientsPage;
+let PatientsListColumns: ColDef[] = [
+  {
+    headerName: "View",
+    cellRenderer: (params: CellClassParams<any>) => {
+      return (
+        <Button
+          variant="outline-secondary"
+          size="sm"
+          onClick={() => {
+            if (params.data.value !== undefined) {
+              params.context.navigateFunction(`/patients/${params.data.value}`);
+            }
+          }}
+        >
+          View
+        </Button>
+      );
+    },
+    sortable: false,
+  },
+  {
+    field: "patientMrn",
+    headerName: "Patient MRN",
+    hide: true,
+    cellStyle: { color: "crimson" },
+  },
+  {
+    field: "cmoPatientId",
+    headerName: "CMO Patient ID",
+    valueGetter: function ({ data }) {
+      for (let i of data["isAliasPatients"][0]["patientAliasesIsAlias"]) {
```

Comment:
never use native for loops.   if your iterating over an array, use array.forEach.   if object, use lodash _.forIn

Reply:
actually, this loop would be more semantically expressed with array.find.  it would also let you handle case where nothing is found.